### PR TITLE
Support xfield for rust-tapyrus

### PR DESCRIPTION
**Pull request by Yamaguchi**
This pull request will add an extra field feature proposed at https://github.com/chaintope/tapyrus-core/issues/66

and a part of solution of the [support tapyrus v0.4.0](https://github.com/chaintope/rust-tapyrus/issues/13)

### CODE REVIEW

File: src/blockdata/block.rs [hunk 1]
```diff
@@ -49,33 +52,137 @@ pub struct BlockHeader {
...
+        match self.extra_field {
+            ExtraField::AggregatePublicKey(pk) => Some(pk),
+            _ => None,
+        }
+    }
+}
+
+/// An extra field that allows the block header to hold arbitrary data.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ExtraField {
```
Comment (rantan) [inline src/blockdata/block.rs]: 👍

File: src/blockdata/block.rs [hunk 2]
```diff
@@ -49,33 +52,137 @@ pub struct BlockHeader {
...
+        S: serde::Serializer,
+    {
+        serializer.collect_str(&self)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> ::serde::Deserialize<'de> for ExtraField {
+    fn deserialize<D: ::serde::Deserializer<'de>>(d: D) -> Result<ExtraField, D::Error> {
+        struct ExtraFieldVisitor;
```
Comment (rantan) [inline src/blockdata/block.rs]: 👍 I didn't know that we can define structs in a function.

**Review by rantan — COMMENTED** [review]
Regarding logic and structure changes are pretty good. 👍 

But, I think the field name should be `xfield` and `xfieldType` for consistency.
What do you think?

File: src/blockdata/block.rs (hunk 1 above)
Comment (Yamaguchi) [reply src/blockdata/block.rs]: Fixed at b15ff3efc64e2748b728888fd7c2a865cc77004c

**Pull request merged**